### DISCUSSION TITLE
configurable prefixes and example queries in fuseki UI (yasgui)

### DISCUSSION THREAD

**Discussion by SimonBin** [apache/jena#1680] [Ideas]
![image](https://user-images.githubusercontent.com/10867832/207991271-df9ed0b7-4ee1-41db-a69f-d555ae675d78.png)

could be nice to have it configurable

I have a sample patch here :

https://github.com/AKSW/jena/commit/ff4af60fcbe8e4f2683a4285e277d898225a846a

however I would like to know if this would be a good feature and how it would be good to implement before I pursue a Pull request

in this patch, you could create a config but the dataset needs to be named yasgui-config, with such a content:

```
PREFIX : <https://yasgui.triply.cc/config#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX sd: <http://www.w3.org/ns/sparql-service-description#>

[
  sd:endpoint "/my-endpoint/" ;
  :exampleQueries (
    [ rdfs:label "spo10" ;
      :query """select * { ?s ?p ?o } limit 10""" ]
    [ rdfs:label "ontology classes" ;
      :query """PREFIX owl: <http://www.w3.org/2002/07/owl#>
select * { ?cls a owl:Class }""" ]
  ) ;
  :prefixes (
    [ rdfs:label "owl" ; :uri <http://www.w3.org/2002/07/owl#> ]
    [ rdfs:label "dct" ; :uri <http://purl.org/dc/terms/> ]
  ) ;
] .
```

**Comment by kinow** [comment]
+1 from me. Not sure if we might want to cache it to avoid accessing the server for every query (maybe have a way in the UI to ask it to be refreshed from the server). But this can be refined later. Thanks!

**Comment by afs** [comment]
Just to be clear:  A PR is best practice - we don't take material from elsewhere when a PR can be done. Random material on the web isn't necessarily "released".

PR content is under the contributor license.

```
JSON {
 "": ?text,
 "": ?value
} WHERE
```

Is using a little-known Jena extension necessary? The query UI should work with any SPARQL endpoint, any compliant triple store, and if read-only or if the target service does not exist.

(and it will break when #1678 is fixed)

**Reply by SimonBin** [reply]
another problem is that the query uses list:index... so the question is how far do we want to take the query UI to work on nonFuseki? The code also expects endpoints such as /$/server to be present...

**Reply by afs** [reply]
Yes but the extension query form JSON is suspect and by not using it, we keep the option to remove it.
It is an experiment, no traction.
GraphQL is the way forward.

(Though IMO having something simple to generate JSON for a result set seems sensible to me. I can imagine the choice of format would cause "much debate" if on the SPARQL CG.)

**Comment by nichtich** [comment]
I created https://github.com/apache/jena/issues/1793 without knowing this discussion. The possible implementation I thought of is configuration via local files. This could also be used to other customization such as title and logo.

As far as what I've seen example queries and prefixes are not part of Yasgui interface so hijacking the https://yasgui.triply.cc/config# namespace would not be ok. Yasr, part of Yasgui supports configuration of prefixes for a different purpose but [configuration is JSON](https://triply.cc/docs/yasgui-api/#yasr-configuration) so I'd stick to plain JSON files and `.rq` files for sample queries.

**Comment by nichtich** [comment]
See https://github.com/gbv/jena/commit/cd5c2e2d21ec385a2b15c2aa2b1da611c21c8320 for a working patch to configure prefixes and example queries via a config file.
